### PR TITLE
Fixed java contribution

### PR DESCRIPTION
**Pull request by akosyakov**
After splitting examples.

### CODE REVIEW

**Comment by kittaakos** [comment]
I will look into this. I have to try it out locally.

File: package.json [hunk 1]
```diff
@@ -54,6 +56,10 @@
     "circular-dependency-plugin": "^2.0.0",
     "copy-webpack-plugin": "^4.0.1",
     "css-loader": "^0.26.2",
+    "find-package-json": "^1.0.0",
```
Comment (kittaakos) [inline package.json]: Where do we use this dependency? Maybe I've just overlooked something.

File: package.json [hunk 2]
```diff
@@ -89,7 +96,10 @@
     "clean:monaco-languageclient": "./node_modules/.bin/rimraf node_modules/monaco-languageclient && ./node_modules/.bin/rimraf node_modules/vscode-base-languageclient",
     "update:monaco-languageclient": "npm run clean:monaco-languageclient && npm install",
     "clean:file-deps": "npm run clean:monaco-languageclient && npm run clean:vscode-ws-jsonrpc",
-    "update:file-deps": "npm run clean:file-deps && npm install"
+    "update:file-deps": "npm run clean:file-deps && npm install",
+    "java:download": "./node_modules/.bin/gulp download_java_server",
+    "java:build": "./node_modules/.bin/gulp build_java_server",
```
Comment (kittaakos) [inline package.json]: Running the `npm run java:build` failed for me. The `npm run java:download` worked fine but with the build, I got an error.

```
[ERROR] Failed to resolve target definition /Users/akos.kitta/git/eclipse.jdt.ls/org.eclipse.jdt.ls.target/org.eclipse.jdt.ls.tp.target: Could not find "org.eclipse.m2e.feature.feature.group/1.8.0.20170329-2113" in the repositories of the current location -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
[14:22:25] 'build_java_server' errored after 1.2 min
[14:22:25] Error: Command failed: ./mvnw -Pserver-distro clean package
    at checkExecSyncError (child_process.js:483:13)
    at Object.execSync (child_process.js:523:13)
    at Gulp.gulp.task (/Users/akos.kitta/git/theia/gulpfile.js:16:8)
    at module.exports (/Users/akos.kitta/git/theia/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/akos.kitta/git/theia/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/akos.kitta/git/theia/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/akos.kitta/git/theia/node_modules/orchestrator/index.js:134:8)
    at /Users/akos.kitta/git/theia/node_modules/gulp/bin/gulp.js:129:20
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)


```

File: package.json (hunk 1 above)
Comment (akosyakov) [inline package.json]: I think it is used by `file-dependency-updater`

Comment (kittaakos) [inline package.json]: Then that is already declared in its `package.json`.

**Pull request merged**